src/expr/src/vector_op/string_to_array.rs

```diff
@@ -17,23 +17,38 @@ use risingwave_common::array::ListValue;
 use risingwave_common::types::ScalarImpl;
 use risingwave_expr_macro::function;
 
-fn string_to_array_inner(s: &str, sep: Option<&str>) -> Vec<String> {
+enum StringIterator<'a> {
```

Comment:
What about having a try for `auto_enums` here?

https://github.com/risingwavelabs/risingwave/blob/26a950fa5716b251af2bc711d78fff64650406b1/src/common/src/array/vis.rs#L184-L190

Reply:
Intersting tool 😁
I believe it does not support nested if-else/map, therefore I separate inner to inner and "inner_inner" to workaround😇

Reply:
Is this what you want? 😄

https://docs.rs/auto_enums/latest/auto_enums/#nested-armsbranches